NAMESPACE

```diff
@@ -1,6 +1,9 @@
 # Generated by roxygen2: do not edit by hand
 
+S3method("$",qenv.error)
+S3method("$<-",qenv.error)
 S3method("[[",qenv.error)
+S3method("[[<-",qenv.error)
```

Comment:
Why is it needed?

Reply:
I'd like to preface that I'm more than happy to remove as this is \*\*totally\*\* nitpicking.

My reasoning adding this would be to make the API consistent

The 2 snippets below shows the \*\***released/main**\*\* behavior and the assignments (`<-`) seem inconsistent with `qenv`. 

It seems odd that with `qenv.error` the script runs a few more commands.

### `qenv` object without errors

``` r
library(teal.code)
q <- qenv() |> within(a <- 2)
q$something <- 2
#> Error in `$<-`(`*tmp*`, something, value = 2): no method for assigning subsets of this S4 class
q[["another"]] <- 3
#> Error in `[[<-`(`*tmp*`, "another", value = 3): [[<- defined for objects of type "S4" only for subclasses of environment

q$something
#> Error in q$something: $ operator not defined for this S4 class
q[["another"]]
#> object 'another' not found
#> NULL
```

<sup>Created on 2024-10-29 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

### `qenv` object when there's an error in `within` code

``` r
library(teal.code)
q <- qenv() |> within(stop())
q$something <- 2
q[["another"]] <- 3

q$something
#> [1] 2
q[["another"]]
#> Error:  
#>  when evaluating qenv code:
#> stop()
```

<sup>Created on 2024-10-29 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Reply:
We don't need to fence all possible methods. There is large set of the methods inherited from `environment`, it is important that officially supported functions are mentioned in the docs of `qenv`. We need to mention inherited working methods but not add them. What I would add though to `qenv.error` is `$.qenv.error`,  `[[.qenv.error`, `as.list.qenv.error`, `ls.qenv.error`

Reply:
`ls.qenv.error` cannot be added, but it will still give an error, so no harm there

Reply:
hmm, there is a difference between throwing error and returning error. `ls(<qenv.error>)` will throw while `names.qenv.error(<qenv.error>)` will return error. The first one needs to be tryCatched while second one not. We need to check if `ls(data())` in `teal` internals doesn't throw because of this (probably not as `teal` is protected against any types of errors)